Automattic/wp-calypso
# Command palette: navigating to anchor links on same page does not work

Closes https://github.com/Automattic/dotcom-forge/issues/5087

## Proposed Changes

This PR adds a function that allows the user to navigate to an anchor link on the same page when a navigation command is executed through the command palette. 

## Testing Instructions

* Navigate to Calypso Live link or pull the changes from this branch
* Navigate to `Settings > Hosting Configuration` on an Atomic site: `/hosting-config/{yourAtomicSite}`
* Press `cmd+k` on Mac or `ctrl+k` on Windows to open the command palette
* Search for the `Manage staging sites` command and execute it
* Confirm that you are being brought to the right section of the page with staging sites
* Try with other commands such as `Change PHP version` or `Change interface style` and confirm that you are being brought to the right section of the page

**To test that navigation from other pages works as expected**

* Navigate `/home/{yourAtomicSite}`
* Press `cmd+k` on Mac or `ctrl+k` on Windows to open the command palette
* Search for the `Manage staging sites` command and execute it
* Confirm that you are being brought to the `/hosting-config/{yourAtomicSite}#staging-site` and that you end up on the right section of the page

Demo:

https://github.com/Automattic/wp-calypso/assets/25575134/979db715-5443-41f0-a511-f117c48153b7


**Notes**:
* This issue is somewhat common - when we navigate from a different page, the link gets refreshed and it executes the anchor link. When we navigate from the same page to an anchor link, the browser does not refresh the URL on the top so `#staging-site` anchor does not get executed. 
* In the solution, I opted for detecting if we are on the same page and then scrolling into view if we are. If not, the logic should remain the same. It is not perfect because it manipulates the DOM directly but it is used only for a couple of commands. I am scrolling into view by calculating the offset height of the navigation header and subtracting that from the final scroll position to ensure the title of the card remains within view (if you simply scroll to the anchor link, the titles of some cards are not visible). There might be better solutions as well so I am open to suggestions. 

## Pre-merge Checklist
